app/bot/openai.go

```diff
@@ -0,0 +1,94 @@
+package bot
+
+import (
+	"context"
+	"fmt"
+	"github.com/sashabaranov/go-openai"
+	"log"
+	"net/http"
+	"strings"
+)
+
+type OpenAIClient interface {
+	CreateChatCompletion(context.Context, openai.ChatCompletionRequest) (openai.ChatCompletionResponse, error)
+}
+
+// OpenAI bot, returns responses from ChatGPT via OpenAI API
+type OpenAI struct {
+	authToken string
+	client    OpenAIClient
+}
+
+// NewOpenAI makes a bot for ChatGPT
+func NewOpenAI(authToken string, httpClient *http.Client) *OpenAI {
+	log.Printf("[INFO] OpenAI bot with github.com/sashabaranov/go-openai")
+	config := openai.DefaultConfig(authToken)
+	config.HTTPClient = httpClient
+
+	client := openai.NewClientWithConfig(config)
+	return &OpenAI{authToken, client}
+}
+
+// Help returns help message
+func (o *OpenAI) Help() string {
+	return genHelpMsg(o.ReactOn(), "Спросите что-нибудь у ChatGPT")
+}
+
+func (o *OpenAI) chatGPTRequest(request string) (response string, err error) {
+	resp, err := o.client.CreateChatCompletion(
+		context.Background(),
+		openai.ChatCompletionRequest{
+			Model: openai.GPT3Dot5Turbo,
+			Messages: []openai.ChatCompletionMessage{
+				{
+					Role:    openai.ChatMessageRoleUser,
+					Content: request,
+				},
+			},
+		},
+	)
+
+	if err != nil {
+		log.Printf("[WARN] failed to make request to ChatGPT '%s', error=%v", request, err)
```

Comment:
returning error and handling it at the same time is confusing and, usually, too much. In case if error returned the logging is a caller responsibility. I.e. here should be just an error wrapped with some context and no logging

Reply:
I moved logging to `OnMessage` func.

My logic: I wanted to reuse this method in future and have auto-logging for errors

Reply:
Yeah, I get the idea. However, having error handling in multiple places is a fundamentally troublesome idea. In 99% of cases, this should either return the error to the caller or handle the error in place. Logging is one of such "handle the error" methods.